homework/list.js

```diff
@@ -1,79 +1,83 @@
 'use strict';
+function ToDoList(){
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector('#note-list-wrapper');
+}
+ToDoList.prototype = {
+    start: function(){
+	var promise = this.fetchList();
+	promise.then(
+	    (data) => {
+		this.updateList(data);
+		this.drawList();
+		if (this._listNoteContent.length !== 0){
+		    this.preloadFirstNote();
+		}
+	    });
+	promise.catch(
+	    (reason) => {
+		throw reason;
+	    }
+	);
```

Comment:
It's unfortunately wrong. You just got trapped by a common mistake. In fact, you should do this if you don't want to use the plain chaining style:

    promise = promise.then(....);
    promise = promise.catch(....);

Please note it's because every time the `then` or `catch` is called, it will create a new instance of `then-able`. So if you do this:

    promise.then(task1)
    promise.then(task2)

It will **NOT** guarantee `task2` is only after `task1`. In fact, they will execute **concurrently**. So you must make sure the `then-able` is updated as I wrote.